poly-commitment/src/srs.rs

```diff
@@ -289,3 +290,37 @@ where
         }
     }
 }
+
+impl<G: CommitmentCurve> SRS<G>
+where
+    <G as CommitmentCurve>::Map: Sync,
+    G::BaseField: PrimeField,
+{
+    /// This function creates SRS instance for circuits with number of rows up to `depth`.
+    pub fn create_parallel(depth: usize) -> Self {
+        let m = G::Map::setup();
+
+        let g: Vec<_> = (0..depth)
+            .into_par_iter()
+            .map(|i| {
+                let mut h = Blake2b512::new();
+                h.update((i as u32).to_be_bytes());
+                point_of_random_bytes(&m, &h.finalize())
+            })
+            .collect();
+
+        const MISC: usize = 1;
+        let [h]: [G; MISC] = array::from_fn(|i| {
+            let mut h = Blake2b512::new();
+            h.update("srs_misc".as_bytes());
+            h.update((i as u32).to_be_bytes());
+            point_of_random_bytes(&m, &h.finalize())
+        });
```

Comment:
Is the call to `array::from_fn` really warranted here? Or is it to be somewhat "future-proof" wrt to the `MISC` constant?

Reply:
tbh I don't know - when I made this change I copied the code from `SRS::create` (further above) and added `.into_par_iter()` to make it parallel 😅